### PR TITLE
fix(typescript-angular): Fix 'date' format transform to 'string'

### PR DESCRIPTION
**Pull request by Xambey**
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

Fix 'date' format type transform to 'string' type

@TiFu (2017/07) @taxpon (2017/07) @sebastianhaas (2017/07) @kenisteward (2017/07) @Vrolijkx (2017/09) @macjohnny (2018/01) @nicokoenig (2018/09) @topce (2018/10) @akehir (2019/07) @petejohansonxo (2019/11)

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java [hunk 1]
```diff
@@ -122,7 +122,7 @@ public AbstractTypeScriptClientCodegen() {
         typeMapping.put("integer", "number");
         typeMapping.put("Map", "any");
         typeMapping.put("map", "any");
-        typeMapping.put("date", "string");
+        typeMapping.put("date", "Date");
```
Comment (macjohnny) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: This should not be changed for alle typescript generators, as Date serialization/deserialization logic may vary.
Moreover, the angular generator does not implement a deserialization logic, so it should remain string

Comment (Xambey) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: @macjohnny Ok, but why a string in that case hasnt ISO format?

Comment (macjohnny) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: The consuming client / the backend needs to generate strings in iso format

Comment (Xambey) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: > The consuming client / the backend needs to generate strings in iso format

maybe then it is worth making a cast to ISO format? I don't mind having a type be a string if it is handled the same as the date format

Comment (Xambey) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: if you do not use ISO, how will you supply the timezone for the date?

Comment (Xambey) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: tell me, please, how can I build openapi-generator-cli for npm? Found and assembled only java version modules/openapi-generator-cli

Comment (Xambey) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: @macjohnny I want to build a fork version, tell me, please, how can I build openapi-generator-cli for npm? Found and assembled only java version. modules/openapi-generator-cli

Comment (macjohnny) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java]: the string is not formatted by the client, it has to be passed by the consumer in the correct ISO format that the backend will understand.

**Pull request closed without merging**